### PR TITLE
18.4.0-XE - Add save installer option

### LINKED ISSUE #279: OracleDatabase/18.4.0-XE project - Option to save installer rpm
The ability to install Oracle Database 18c XE without having to click through the license agreement is terrific. However, installing directly from OTN means that the installer rpm is downloaded each time the VM is built. The rpm is 2.3 GB, so this can be problematic for people with slow or metered connections.

It might be helpful to add an option to save the installer rpm. This could be added as a Boolean in `.env` and the `Vagrantfile`, with the value passed to `install.sh` as an environment variable. The download/install part of `install.sh` might look something like:

```bash
# Install Oracle
if [[ ! -f /vagrant/oracle-database-xe-18c-1.0-1.x86_64.rpm ]]; then
  echo 'INSTALLER: Downloading Oracle Database software'
  wget -q -P /vagrant \
    https://download.oracle.com/otn-pub/otn_software/db-express/oracle-database-xe-18c-1.0-1.x86_64.rpm
fi

yum -y localinstall /vagrant/oracle-database-xe-18c-1.0-1.x86_64.rpm

if [[ "${KEEP_DB_INSTALLER}" == 'false' ]]; then
  rm -f /vagrant/oracle-database-xe-18c-1.0-1.x86_64.rpm
fi
```

To avoid changing the current behavior of the project, it probably makes sense for KEEP_DB_INSTALLER to default to false.

If you think that this is a good idea, I'd be happy to work on it.

### PR DESCRIPTION
**Pull request by PaulNeumann**
Add an option to the OracleDatabase/18.4.0-XE project to save the database installer file. Changes are:

`.env`
- Add a Boolean configuration parameter called "VM_KEEP_DB_INSTALLER" (default false).

`README.md`
- Document the "VM_KEEP_DB_INSTALLER" parameter in the Configuration section.

`Vagrantfile`
- Add code to evaluate the "VM_KEEP_DB_INSTALLER" parameter and pass it to `install.sh`.

`scripts/install.sh`
- Ensure wget is installed (it's already installed in the current oraclelinux/7 box, but just in case).
- Download the database installer file to `/vagrant`, if it's not already present.
- Install the database software from `/vagrant`, instead of directly from OTN.
- Delete the database installer file if the "KEEP_DB_INSTALLER" environment variable is false.

Closes #279.

To validate:
- Verify that the database installer RPM file is downloaded to `/vagrant` during the build, if it's not already present.
- Verify that the database installer RPM file is deleted if the "VM_KEEP_DB_INSTALLER" parameter is false.
- Verify that the database installer RPM file is kept if the "VM_KEEP_DB_INSTALLER" parameter is true.

I'm happy to make any changes that you'd like.

Signed-off-by: Paul Neumann <38871902+PaulNeumann@users.noreply.github.com>

### CODE REVIEW

File: OracleDatabase/18.4.0-XE/scripts/install.sh [hunk 1]
```diff
@@ -19,6 +19,9 @@ echo 'INSTALLER: Started up'
 # get up to date
 yum upgrade -y
 
+# ensure wget is installed for downloading database installer
+yum install -y wget
```
Comment (Djelibeybi) [inline OracleDatabase/18.4.0-XE/scripts/install.sh:23]: Instead of installing `wget`, perhaps consider using `curl` which is installed by default?

Comment (PaulNeumann) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: @Djelibeybi `wget` is already installed, too. I just didn't want the script to break if that changed in a future version of the base box.

Comment (Djelibeybi) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: Ha, fair enough. Goes to show I should probably do a _modicum_ of verification before reviewing things. :) Good job.

Comment (AmedeeBulle) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: While I used `wget` in the past, I now prefer using `curl` -- `yum` (and many others) depends on `curl`, so you can rely on the fact it is already installed as @Djelibeybi says...

@PaulNeumann: don't change anything, this is just an FYI

Comment (Djelibeybi) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: Right. I switched from `yum` to `curl` after developing the `oraclelinux:7-slim` image (that doesn't have `wget`).

File: OracleDatabase/18.4.0-XE/Vagrantfile [hunk 2]
```diff
@@ -150,6 +157,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.provision "shell", path: "scripts/install.sh", env:
     {
        "SYSTEM_TIMEZONE"     => VM_SYSTEM_TIMEZONE,
+       "KEEP_DB_INSTALLER"   => VM_KEEP_DB_INSTALLER,
```
Comment (AmedeeBulle) [inline OracleDatabase/18.4.0-XE/Vagrantfile:160]: FYI -- while this is correct, you need to keep in mind that `VM_KEEP_DB_INSTALLER` is a boolean not a string.
We have implicit conversion as Ruby booleans are objects implementing the `to_s()` method.

File: OracleDatabase/18.4.0-XE/scripts/install.sh (hunk 1 above)
Comment (PaulNeumann) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: @Djelibeybi @AmedeeBulle Thank you. It sounds like using `curl` would be preferable, so I'd be happy to do that. The download code would then be (tested and ShellChecked):

```bash
if [[ ! -f /vagrant/"${db_installer}" ]]; then
  echo 'INSTALLER: Downloading Oracle Database software'
  (
    cd /vagrant || exit 1
    curl -L -O -s https://download.oracle.com/otn-pub/otn_software/db-express/"${db_installer}"
  )
fi
```

File: OracleDatabase/18.4.0-XE/Vagrantfile (hunk 2 above)
Comment (PaulNeumann) [reply OracleDatabase/18.4.0-XE/Vagrantfile:160]: @AmedeeBulle Understood. Thanks!

File: OracleDatabase/18.4.0-XE/scripts/install.sh (hunk 1 above)
Comment (Djelibeybi) [reply OracleDatabase/18.4.0-XE/scripts/install.sh:23]: Works for me!

Comment (PaulNeumann) [reply OracleDatabase/18.4.0-XE/scripts/install.sh]: Done. Thanks again!

**Review by gvenzl — APPROVED** [review]

**Comment by gvenzl** [comment]
Thanks @PaulNeumann, great contribution, as always! :)

**Pull request merged**